bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/Button.java

```diff
@@ -38,12 +40,26 @@ default String getText() {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * Returns the button link.
+     *
+     * @return the button link
+     * @since com.adobe.cq.wcm.core.components.models 12.10.0
+     */
+    @JsonIgnore  // avoid duplicate URL in JSON, keep old property for backward compatibility
+    default @NotNull Link getButtonLink() {
+        // fallback to old method name for backwards compatibility
+        return new LinkImpl(getLink());
```

Comment:
I would throw instead an `UnsupportedOperationException`. As it is a new method, backwards compatibility does not apply.
This applies to all the models below.

Reply:
when a user has overlayed the sling model but not the markup of a model it works out-of-the box with this design.

if we just throw unsupported all models will break until this new method is implemented as well.

i absolutely dislike this "throw unsupportedoperationexception" pattern, it makes it much more complicated for users who overlay/extend part of the models doing their job, as they are not "notified" by the compiler/in their IDE of new interface additions. but this would be a separate topic and separate issue.

for this PR i think it is a great benefit for the users gradually upgrading from core component version to another, and does not harm anything.

Reply:
> if we just throw unsupported all models will break until this new method is implemented as well.

What do you mean with "break"? Do you mean that it won't build? Adding a new method to a model should not break anything. I would not return a default value in the Model. If needed I would do it in the implementing class.

Reply:
break example:
* we release a new core component version with this PR that introduces the new get*Link methods and updated markup that references it
* assume a customer projects that have added their own implementation of the sling model of a component using the extender pattern, but did not overlay the markup file.
* when the customer deploys the latest version of the core components he is not aware of this new method as the build does not break
* but the markup is broken, as it references the new get*Link method, which just leads to an "unsupportedoperationexception" unless the customer detects the new method and adds an delegation implementation to this own model.
* he can only detect this new method by looking very closely at the diffs between the two core component releases as in the build and IDE everything seems fine because a default implementation in the impl exists (which if it just throws unsupportedoperationexception is a really unhelpful default implementation)

with my proposal of default implementation the use case described is covered and all runs as expected without further changes.

Reply:
Another option to not break backwards compatibility is to create a new version of the  components affected by this change.
I'm not keen with defining a default value in a new method of the Model that relies on a deprecated method.

Reply:
creating a new version of all link-related components would be totally ok for me and a very clean solution - but it did not happen very often until so, so i'm not sure when it's the right time for your team to create a new version (and probably have more maintenance effors with maintaining bugfixes etc. for multiple component versions).

please decide in your team whether you want to create new model version for all link-related components or not. if not i strongly recommend this approach as it's a very simple but helpful solution for the users of this project.